File: csharp/fraud/DeclineEmail.cs
using System;
using System.Reflection;
using com.ultracart.admin.v2.Api;
using com.ultracart.admin.v2.Model;

namespace SdkSample.fraud
{
    public class DeclineEmail
    {
        public static void Execute()
        {
            Console.WriteLine("--- " + MethodBase.GetCurrentMethod()?.DeclaringType?.Name + " ---");
            try
            {
                // declineEmail is a shortcut for telling UltraCart to decline orders from a specific
                // email address. It is the quick alternative to building a full "address email" fraud
                // rule by hand.
                FraudApi fraudApi = new FraudApi(Constants.ApiKey);

                FraudDeclineEmailRequest declineRequest = new FraudDeclineEmailRequest();
                declineRequest.Email = "chargeback-charlie@example.com";

                fraudApi.DeclineEmail(declineRequest);

                Console.WriteLine("Declined email: " + declineRequest.Email);
            }
            catch (Exception e)
            {
                Console.WriteLine("Exception: " + e.Message);
            }
        }
    }
}

File: java/src/fraud/DeclineEmail.java
package fraud;

import com.ultracart.admin.v2.FraudApi;
import com.ultracart.admin.v2.models.FraudDeclineEmailRequest;
import com.ultracart.admin.v2.util.ApiException;
import common.Constants;

/**
 * declineEmail is a shortcut for telling UltraCart to decline orders from a specific email
 * address. It is the quick alternative to building a full "address email" fraud rule by hand.
 */
public class DeclineEmail {
    public static void execute() {
        System.out.println("--- DeclineEmail ---");
        try {
            FraudApi fraudApi = new FraudApi(Constants.API_KEY);

            FraudDeclineEmailRequest declineRequest = new FraudDeclineEmailRequest();
            declineRequest.setEmail("chargeback-charlie@example.com");

            fraudApi.declineEmail(declineRequest);

            System.out.println("Declined email: " + declineRequest.getEmail());
        } catch (ApiException e) {
            System.out.println("Exception: " + e.getMessage());
        }
    }
}
